documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1777,3 +1778,28 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 **How to fix the violation**: Provide detailed description of the node in the description section.
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R4001" ></a>R4001 XmsParameterLocation
+
+**Category** : SDK Error
+
+**Applies to** : ARM and Data Plane OpenAPI(swagger) specs
+
+**Output Message** : The parameter 'managementGroupId' is defined in global parameters section without 'x-ms-parameter-location' extension. This would add the parameter as the client property. Please ensure that this is exactly you want. If so, apply the extension "x-ms-parameter-location": "client". Else, apply the extension "x-ms-parameter-location": "method".
+
+**Description** : By default Autorest processes global parameters as properties on the client. For example subscriptionId and apiVersion which are defined in the global parameters section end up being properties of the client. It would be natural to define resourceGroupName once in the global parameters section and then reference it everywhere, rather than repeating the same definition inline everywhere. One may not want resourceGroupName as a property on the client, just because it is defined in the global parameters section. This extension helps you achieve that. You can add this extension with value "method" "x-ms-parameter-location": "method" and resourceGroupName will not be a client property.
```

Comment:
Same, it would be nice to be a little more clear here about the difference between the two options.

Maybe "In contrast, resourceGroupName is also defined in the global parameters section to avoid repeating the same definition inline everywhere, but it's intended to be provided to each method rather than exposed as a property of the client itself."

Reply:
Truthfully, this was copy/pasted from here:
[Azure/autorest@dbc4571 docs/extensions#x-ms-parameter-location](https://github.com/Azure/autorest/tree/dbc4571f5b4e3142e4cae5370e3cbbe00dea1af0/docs/extensions#x-ms-parameter-location)

However, I agree with your criticism and will rewrite this. :) Thanks for keeping me honest!

Reply:
Somehow I knew the poor quality of this prose was not your fault :p